Dockerfile

```diff
@@ -1,19 +1,15 @@
 # Build the manager binary
 FROM quay.io/centos/centos:stream8 AS builder
-RUN dnf install -y golang git \
-    && dnf clean all -y
 
 WORKDIR /workspace
-# Copy the Go Modules manifests
+# Copy the Go Modules manifests for detecting Go version
 COPY go.mod go.mod
 COPY go.sum go.sum
 
-# Ensure correct Go version
-RUN export GO_VERSION=$(grep -E "go [[:digit:]]\.[[:digit:]][[:digit:]]" go.mod | awk '{print $2}') && \
-    go install golang.org/dl/go${GO_VERSION}@latest && \
-    ~/go/bin/go${GO_VERSION} download && \
-    /bin/cp -f ~/go/bin/go${GO_VERSION} /usr/bin/go && \
-    go version
+# Install Golang based on the version from go.mod
+RUN GO_VERSION=$(grep -E "go [[:digit:]]\.[[:digit:]][[:digit:]]" go.mod | awk '{print $2}') \
+    && dnf install golang-1.21.7 git -y && go version \
```

Comment:
Why extract GO_VERSION if we always install `1.21.7`?

Reply:
that was for testing :(

Reply:
The idea of using the patch version in go.mod isn't supported in Golang v1.20 and below, so I will introduce it in #133